### ISSUE TITLE
Parantheses tree: Only escape protected character

### ISSUE THREAD

**Issue by jenetics** [jenetics/jenetics#481]
Currently also the '\' is escaped for the parentheses tree, no matter whether it follows a protected character ('(', ')' or ','). This makes it quite cumbersome if the node value itself also contains an escaped character sequence.

Before: escape("a\\") -> "a\\\\", escape("\\a") -> "\\\\a"
After: escape("a\\") -> "a\\", escape("\\a") -> "\\a"

**Comment by jenetics** [comment]
Merged into r5.0.0 branch.
